### ISSUE TITLE
Descriptor `ClassVar` results in false positive when assigned through class instance

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4838]
```python
from typing import ClassVar, Generic, Self, TypeVar, overload

T = TypeVar("T")

class Field(Generic[T]):
    @overload
    def __get__(self, instance: None, owner) -> Self:
        ...

    @overload
    def __get__(self: Field[T], instance: object, owner) -> T:
        ...

    def __get__(self, instance: object | None, owner) -> Self | T:
        ...

    def __set__(self, instance: object, value: T) -> None:
        ...

    def is_null(self) -> bool:
        ...


class StrField(Field[str]): ...

class Example:
    name: ClassVar = StrField()

    def reset(self) -> None:
        self.name = "" # False positive here
```

**Comment by erictraut** [comment]
This will be addressed in the next release.
